### PR TITLE
Update abilities.js

### PR DESCRIPTION
**Pull request by charizard8888**
Merge it

### CODE REVIEW

**Comment by charizard8888** [comment]
Merge + Updateserver + Hotopatch @Spandamn

**Comment by charizard8888** [comment]
Ignore tab spacing in learnsets

**Comment by Spandamn** [comment]
Won't merge until the you change what I've requested

**Comment by charizard8888** [comment]
What changes?

**Comment by charizard8888** [comment]
Boi too many changes

File: mods/thefirstnewgen/learnsets.js [hunk 1]
```diff
@@ -422,4 +422,599 @@ exports.BattleLearnsets = {
...
+                fakeout: ["5L1"],
+                perishsong: ["5L1"],
+                irondefense: ["5L1"],
+                harden: ["5L1"],
+                wildcharge: ["5L1"],
+                flareblitz: ["5L1"],
+                waterpledge
+                },
+
+dontidahl: {
```
Comment (charizard8888) [inline mods/thefirstnewgen/learnsets.js]: Ignore the learnsets hmm

File: mods/thefirstnewgen/moves.js [hunk 2]
```diff
@@ -1148,4 +1148,37 @@ exports.BattleMovedex = {
...
+		basePower: 0,
+		category: "Status",
+		shortDesc: "Infects the target.",
+		id: "infection",
+		isViable: true,
+		name: "Infection",
+		pp: 10,
+		priority: 0,
+		flags: {protect: 1, reflectable: 1, mirror: 1},
+		// No Guard-like effect for Poison-type users implemented in BattleScripts#tryMoveHit
```
Comment (charizard8888) [inline mods/thefirstnewgen/moves.js]: Fixed this

File: mods/thefirstnewgen/abilities.js [hunk 3]
```diff
@@ -41,15 +56,33 @@ exports.BattleAbilities = {
...
 				return this.chainModify(1.5);
 			}
 		},
+				onSourceModifyAtk: function (atk, attacker, defender, move) {
+			if (move.type === 'Fire' || move.type === 'Water' || move.type === 'Electric' || move.type === 'Grass') {
+				this.debug('Draconic Light weaken');
+				return this.chainModify(0.5);
+			}
+		},
+		onModifySpAPriority: 5,
```
Comment (charizard8888) [inline mods/thefirstnewgen/abilities.js]: So remove all this?

Comment (charizard8888) [inline mods/thefirstnewgen/abilities.js]: I dont get this one :P

File: mods/thefirstnewgen/abilities.js [hunk 4]
```diff
@@ -41,15 +56,33 @@ exports.BattleAbilities = {
...
+	"draconiclight": { 
+		shortDesc: "1.5x Dragon moves and 1/2 damage from fire, water, electric, and grass, but 2x from Dragon",
 		onBasePowerPriority: 8,
 		onBasePower: function(basePower, attacker, defender, move) {
 			if (move.type === 'Dragon') {
 				this.debug('DraconicLight boost');
 				return this.chainModify(1.5);
 			}
 		},
+				onSourceModifyAtk: function (atk, attacker, defender, move) {
```
Comment (charizard8888) [inline mods/thefirstnewgen/abilities.js]: Spaced properly, idk how to do that in onSOurceModifyDamage though :P

**Comment by charizard8888** [comment]
@Spandamn I did most of the stuff and dunno how to do the rest so you do all that plz :]

File: mods/thefirstnewgen/learnsets.js (hunk 1 above)
Comment (Spandamn) [inline mods/thefirstnewgen/learnsets.js]: Just run the through jsbeautifier ez

File: mods/thefirstnewgen/abilities.js (hunk 3 above)
Comment (Spandamn) [inline mods/thefirstnewgen/abilities.js]: Yes remove this

**Comment by charizard8888** [comment]
Commenting on all the stuff that you have ta da naw !

File: mods/thefirstnewgen/abilities.js [hunk 5]
```diff
@@ -41,15 +56,33 @@ exports.BattleAbilities = {
...
+			if (move.type === 'Fire' || move.type === 'Water' || move.type === 'Electric' || move.type === 'Grass') {
+				this.debug('Draconic Light weaken');
+				return this.chainModify(0.5);
+			}
+		},
+		onModifySpAPriority: 5,
+		onSourceModifySpA: function (atk, attacker, defender, move) {
```
Comment (charizard8888) [inline mods/thefirstnewgen/abilities.js]: You do this one

File: mods/thefirstnewgen/learnsets.js (hunk 1 above)
Comment (charizard8888) [inline mods/thefirstnewgen/learnsets.js]: Okek will do but probs won't get time on pc soon

**Comment by charizard8888** [comment]
@Spandamn hotpatch when you're back, gotta test something (hotpatch even if this is not merged)

File: mods/thefirstnewgen/abilities.js (hunk 3 above)
Comment (charizard8888) [inline mods/thefirstnewgen/abilities.js]: You sure? Read the desc and if still yes you do et cuz I won't be on PC till hmm tomorrow nite

**Comment by charizard8888** [comment]
Wew this PR is interesting

**Pull request merged**